enaml/src/funchelper.cpp

```diff
@@ -20,18 +20,19 @@ from Python's funcobject.c
 
 */
 PyObject*
-call_func( PyObject* mod, PyObject* args )
+call_func( PyObject* mod, PyObject *const *args, Py_ssize_t nargs )
 {
-    PyObject* func;
-    PyObject* func_args;
-    PyObject* func_kwargs;
-    PyObject* func_locals = Py_None;
-
-    if( !PyArg_UnpackTuple( args, "call_func", 3, 4, &func, &func_args, &func_kwargs, &func_locals ) )
+    if( !(nargs == 4 || nargs == 3) )
     {
+        PyErr_SetString( PyExc_TypeError, "call_func must have 3 or 4 arguments" );
         return 0;
     }
 
+    PyObject* func = args[0];
```

Comment:
This is not the proper way to access the items in a tuple.

You should use `PyTuple_GET_ITEM`, which returns a borrowed reference.

Reply:
I thought the point of using [METH_FASTCALL](https://docs.python.org/3/c-api/structures.html#c.PyCFunctionFast) was to avoid packing/unpacking tuples?

Reply:
I stand corrected. Hadn't seen that convention before. Sorry for the noise!